docs/server-config.md

```diff
@@ -4,47 +4,146 @@
-   - i.e. Running the command:
+Configuration parameters can be updated with:
 
-     ```sh
-     python tabpy.py --config=path\to\my\config
-     ```
+1. Adding environment variables - set the environment variable as required by
+   your Operating System. When creating environment variables, use the same
+   name as is in the config file as an environment variable.
+2. Specifying a parameter in a config file (enviroment variable value overwrites
+   configuration setting).
+
+Configuration file with custom settings is specified as a command line parameter:
+
+```sh
+tabpy --config=path/to/my/config/file.conf
+```
 
 The default config file is provided to show you the default values but does not
 need to be present to run TabPy.
 
+### Configuration File Content
+
+Configuration file consists of settings for TabPy itself and Python logger
+settings. You should only set parameters if you need different values than
+thr defaults.
+
+TabPy parameters explained below and for the logger documentation
+can be found at [`logging.config` documentation page](https://docs.python.org/3.6/library/logging.config.html).
+
+`[TabPy]` parameters:
+
+- `TABPY_PORT` - port for TabPy to listen on. Default value - `9004`.
+- `TABPY_QUERY_OBJECT_PATH` - query objects location. Used with models, see
+  [TabPy Tools documentation](tabpy-tools.md) for details. Default value -
+  `/tmp/query_objects`.
+- `TABPY_STATE_PATH` - state location for Tornado web server. Default
+   value - `tabpy/tabpy_server` subfolder in TabPy package folder.
+- `TABPY_STATIC_PATH` - location of static files (index.html page) for
+  TabPy instance. Default value - `tabpy/tabpy_server/static` subfolder in
+  TabPy package folder.
+- `TABPY_PWD_FILE` - path to password file. Setting up this parameter
+  makes TabPy require credentials with HTTP(S) requests. More details about
+  authentication can be found in [Authentication](#authentication)
+  section. Default value - not set.
+- `TABPY_TRANSFER_PROTOCOL` - transfer protocol. Default value - `http`. If
+  set to `https` two additional parameters have to be specified:
+  `TABPY_CERTIFICATE_FILE` and `TABPY_KEY_FILE`. More details for how to
+  configure TabPy for HTTPS are at [Configuring HTTP vs HTTPS]
+  (#configuring-http-vs-https) section.
+- `TABPY_CERTIFICATE_FILE` to certificate file to run TabPy with. Only used
+  with `TABPY_TRANSFER_PROTOCOL` set to `https`. Default value - not set.
+- `TABPY_KEY_FILE` to private key file to run TabPy with. Only used
+  with `TABPY_TRANSFER_PROTOCOL` set to `https`. Default value - not set.
+- `TABPY_LOG_DETAILS` - when set to `true` additional call information
+  (caller IP, URL, client info, etc.) is logged. Default value - `false`.
+- `TABPY_EVALUATE_TIMEOUT` - script evaluation timeout in seconds. Default
+  value - `30`.
+
+### Configuration File Example
+
+```toml
+[TabPy]
+# TABPY_QUERY_OBJECT_PATH = /tmp/query_objects
+# TABPY_PORT = 9004
+# TABPY_STATE_PATH = ./tabpy/tabpy_server
```

Comment:
Let's document that these should be absolute paths not relative paths.

Reply:
fixed